lib/spack/spack/new_installer.py

```diff
@@ -1078,6 +1078,19 @@ def _enqueue_parents(self, dag_hash: str) -> None:
                 self.pending_builds.append(parent)
 
     def install(self) -> None:
+        # This installer has not implemented the per-spec exclusive locks during installation.
+        # Instead, take an exclusive lock on the entire range to avoid that other Spack install
+        # process start installing the same specs.
+        lock = spack.util.lock.Lock(
+            str(spack.store.STORE.prefix_locker.lock_path), desc="prefix lock"
```

Comment:
I think this needs `length=sys.maxsize` to ensure that another process cannot grab a prefix lock on a location inside the same file.

The default is `start=0, length=0`, which I believe means that any byte except for the first byte of the file could be acquired by another process while this lock is held.